tests/testutil/service.go

```diff
@@ -202,6 +203,98 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+// GetEndpointURL returns the URL for this service's app if the App object is versioned it returns
+// the versioned url.
+func (ts *TestService) GetEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+// GetKubeEndpointUrlByRevision returns the versioned endpoint URL by using the revision provided
+// and returns it as string
+func (ts *TestService) GetKubeEndpointURL() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	args := []string{"--format", "json",
+		fmt.Sprintf("%s:%s",
+			strings.Split(ts.Name, ":")[0],
+			strings.Split(ts.Name, ":")[1])}
+	out, err := RioCmd("inspect", args)
+	s := riov1.Service{}
+	if err := json.Unmarshal([]byte(out), &s); err != nil {
+		return ""
+	}
+	args = []string{"get", "service.rio.cattle.io",
+		"-n", testingNamespace,
+		s.Name,
+		"-o", `jsonpath="{.status.endpoints[0]}"`}
+	url, err := KubectlCmd(args)
```

Comment:
Whats the point of the first Riocmd call and unmarshall ? Can't you just delete that entirely and use ts.Name ?

Reply:
I'm getting the actual name of the revision with rio, then I feed kubectl to get the endpoint. @StrongMonkey's suggestion.

Reply:
You should be able to just use `ts.name` or `ts.version`, really anything you get from the unmarshall already exists on the `ts` object.

Reply:
using `ts.Service.Name`